tests/functional/s3/test_sync_command.py

```diff
@@ -353,6 +353,64 @@ def test_with_copy_props(self):
             ]
         )
 
+    def test_upload_with_checksum_algorithm_sha1(self):
+        self.files.create_file('foo.txt', 'contents')
+        cmdline = f'{self.prefix} {self.files.rootdir} s3://bucket/ --checksum-algorithm SHA1'
+        self.run_cmd(cmdline, expected_rc=0)
+        self.assert_in_operations_called(
+            ('PutObject', {
+                'Bucket': 'bucket',
+                'Key': 'foo.txt',
+                'ChecksumAlgorithm': 'SHA1',
+                'Body': mock.ANY,
+                'ContentType': 'text/plain'
+            })
+        )
+
+    def test_upload_with_checksum_algorithm_sha256(self):
+        self.files.create_file('foo.txt', 'contents')
+        cmdline = f'{self.prefix} {self.files.rootdir} s3://bucket/ --checksum-algorithm SHA256'
+        self.run_cmd(cmdline, expected_rc=0)
+        self.assert_in_operations_called(
+            ('PutObject', {
+                'Bucket': 'bucket',
+                'Key': 'foo.txt',
+                'ChecksumAlgorithm': 'SHA256',
+                'Body': mock.ANY,
+                'ContentType': 'text/plain'
+            })
+        )
+
+    def test_download_with_checksum_mode_sha1(self):
+        self.parsed_responses = [
+            self.head_object_response(),
+            {
+                'ContentLength': '100',
+                'LastModified': '00:00:00Z',
+                'ETag': 'foo-1',
+                'ChecksumSHA1': 'checksum',
+                'Body': BytesIO(b'foo')
+            }
+        ]
+        cmdline = f'{self.prefix} s3://bucket/foo {self.files.rootdir} --checksum-mode ENABLED'
+        self.run_cmd(cmdline, expected_rc=0)
+        self.assertEqual(self.operations_called[0][0].name, 'ListObjectsV2')
```

Comment:
I ran the command manually with debug and verified that a download sync does not call GetObject. I also added a print statement to output all operations called when running this test and ListObjectV2 was the only one.

Reply:
But not having `GetObject` defeats the purpose of this test since we're explicitly testing the `ChecksumMode` parameter with `sync`'s download feature. It looks like the issue here is that we have `self.head_object_response()` as the stubbed response for `ListObjectsV2` when it should be `self.list_objects_response(keys['foo-1'])` instead.